infrastructure/tests/test_data_import.py

```diff
@@ -0,0 +1,115 @@
+from django.test import TestCase, Client, override_settings, TransactionTestCase
+from django.conf import settings
+from django.urls import reverse
+from io import BytesIO
+from django.contrib.auth.models import User
+from django.contrib.admin.sites import AdminSite
+from django_q.models import OrmQ
+from django_q.tasks import AsyncTask, fetch, result
+import time
+import rest_framework.response
+
+from infrastructure.models import FinancialYear, QuarterlySpendFile, AnnualSpendFile, Expenditure, Project
+from infrastructure.tests import utils
+from infrastructure.utils import load_excel
+from scorecard.models import Geography
+
+
+@override_settings(Q_CLUSTER={**settings.Q_CLUSTER, 'sync': True, 'poll': 2, 'max_attempts': 3})
+class FileTest(TransactionTestCase):
+
+    def setUp(self):
+        fixtures = ["seeddata.json"]
+        self.client = Client()
+        self.username = 'admin'
+        self.email = 'test@whatever.com'
+        self.password = 'password'
+        self.user = User.objects.create_superuser(self.username, self.email, self.password)
+        FileTest.geography = Geography.objects.create(
+            geo_level="municipality",
+            geo_code="BUF",
+            province_name="Eastern Cape",
+            province_code="EC",
+            category="A",
+        )
+
+    def test_file_upload(self):
+        """Scope of Test: Testing the file upload in Django Admin to processing file and add to Django_Q"""
+        self.client.login(username=self.username, password=self.password)
+        fy = FinancialYear.objects.create(budget_year="2019/2020", active=1)
+
+        self.assertEquals(FinancialYear.objects.all().count(), 1)
+        self.assertEquals(AnnualSpendFile.objects.all().count(), 0)
+
+        # the app name, the name of the model and the name of the view
+        url = reverse('admin:infrastructure_annualspendfile_add')
+
+        with open('infrastructure/tests/test_files/test.xlsx', 'rb', ) as f:
```

Comment:
This file should be minimal. I think it should have 2 projects for 2 municipalities.